components/formats-gpl/src/loci/formats/in/ZeissCZIReader.java

```diff
@@ -715,7 +715,7 @@ protected void initFile(String id) throws FormatException, IOException {
       seriesCount *= mosaics;
     }
 
-    ms0.imageCount = getSizeZ() * (isRGB() ? 1 : getSizeC()) * getSizeT();
+    ms0.imageCount = getSizeZ() * (isRGB() ? getSizeC()/3 : getSizeC()) * getSizeT();
```

Comment:
/**
   * Gets the effective size of the C dimension, guaranteeing that
   * getEffectiveSizeC() * getSizeZ() * getSizeT() == getImageCount()
   * regardless of the result of isRGB().
   */
  int getEffectiveSizeC();

Thus, getEffectiveSizeC() is basically the whole of (isRGB() ? getSizeC()/3 : getSizeC()) and I guess it is calculated later, once the imageCount is set at this level. This is how I think it works, but I could be wrong. @melissalinkert @sbesson does that sound right?